support Bootstrap 4

afeld opened the pull request:
Closes #39.

* [x] Upgrade Bootstrap in demo site
* [x] Switch from using Affix component ([support was dropped](https://v4-alpha.getbootstrap.com/migration/#components))
* [x] Fix ScrollSpy
* [ ] ~~Ensure backwards-compatibility (create a separate test page for Bootstrap 3)~~

afeld committed:
upgrade to Bootstrap 4

alcalyn commented:
Links in navigation was inline, so I needed to add:

``` css
nav[data-toggle='toc'] .nav {
  flex-direction: column;
}
```

Commit:
Updated bootstrap to 4.0

Changed css and js cdn version to bootstrap/4.0.0/

Commit:
Affix -> Sticky

Replaced affix bootstrap attributes with position: sticky; css attribute.
Also updated index.md documentation to reflect this

Commit:
Fix ScrollSpy

- Added navlink class to <a> tags
- Modified selectors to select .active .nav-link elements and their sibling ul elements
- Added navbar-nav to <ul> tags to ensure vertical align

Known issue:
- On the test templates, scrollspy does not highlight the last entry when you scroll to the bottom of the page. This behaviour was not present previously

Commit:
Updated test cases to include new classes

lokimckay commented:
Completed steps 1-3 through path of least resistance
PR: #43

afeld committed:
Merge pull request #43 from sneyed/bootstrap-4

#40 Bootstrap 4 - Steps 1-3

afeld commented on _layouts/default.html:
```diff
@@ -8,12 +8,13 @@
-    <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
-    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js" integrity="sha512-K1qjQ+NcF2TYO/eI3M6v8EiNYZfA95pQumfvcVrTHtwQVDG+aHRqLi/ETn2uB+1JqwYqVG3LIvdm9lj6imS/pQ==" crossorigin="anonymous"></script>
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.11.0/umd/popper.min.js" integrity="sha384-b/U6ypiBEHpOf/4+1nzFpr53nxSS+GLCkfwBdFNTxtclqqenISfwAzpKaMNFNmj4" crossorigin="anonymous"></script>
```
Are we actually using this?

lokimckay replied:
- Have not used popper personally, but from a quick google it doesn't look like it is being used anywhere
- jQuery is definitely being used

afeld committed:
point users to old instructions for Bootstrap v3

afeld committed:
leverage .sticky-top from Bootstrap

afeld committed:
fix second-level nav jumping right when it becomes active

afeld committed:
use Scrollspy links for Bootstrap v4

afeld committed:
remove unused Popper.js script

afeld committed:
move layout positioning to be site-specific

afeld commented:
I think this is good to go. @sneyed Mind giving it another look, just to make sure I didn't miss anything?

lokimckay commented:
Looks great to me except for one issue: when sticking, there is no gap between the top and toc (see image)
https://user-images.githubusercontent.com/14341953/37190429-60e93bc2-23a5-11e8-9c65-59c8bc65cc35.png

Are you happy with this? I am thinking we should add a small amount of margin or padding - not sure which.

RishikeshDarandale commented:
@afeld, when this will be available as a release version?

afeld commented:
As soon as I can get back to it and figure out why there's a discrepancy in what @sneyed and I are seeing.

RishikeshDarandale commented:
Okay. `.stick-top` has `top` as 0, thus there is no gap. Probably below should work:
```
.sticky-top {
  top: 42px;
}
```

afeld committed:
ensure that the example layout CSS and the styles used for the site stay in sync

afeld committed:
fix sticky navigation vertical positioning

afeld commented:
Ok, fixed, and changed the site to share styles between the documentation and the site itself. Mind taking another look?

RishikeshDarandale commented:
Thanks! Looks great! For reference check [here](https://rishikeshdarandale.github.io/aws-http/).

afeld committed:
add note about upgrading

lokimckay commented:
Thanks Aidan, great work :+1:

afeld commented:
Released as v1.0.0. Thanks everyone!
